runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -400,14 +420,13 @@ private ILogData cacheLoadAndGet(Cache<Long, ILogData> cache, long address, ILog
      * Read the given object from a range of addresses.
      *
      * @param addresses An iterable with addresses to read from
-     * @param waitForWrite Flag whether wait for write is required or hole fill directly.
      * @return A map of addresses read, which will be cached if caching is enabled
      */
-    public Map<Long, ILogData> read(Iterable<Long> addresses, boolean waitForWrite, boolean ignoreTrimmed) {
+    public Map<Long, ILogData> read(Iterable<Long> addresses, ReadOptions options) {
```

Comment:
Also can we annotate readOption with NonNull

Reply:
Done.